framework/mesh/MeshGenerator/SplitFileMeshGenerator.cc

```diff
@@ -0,0 +1,398 @@
+#include "SplitFileMeshGenerator.h"
+
+#include "data_types/byte_array.h"
+#include "utils/chi_utils.h"
+
+#include "mesh/MeshHandler/chi_meshhandler.h"
+#include "mesh/VolumeMesher/chi_volumemesher.h"
+#include "mesh/MeshContinuum/chi_meshcontinuum.h"
+
+#include "chi_log.h"
+#include "utils/chi_timer.h"
+
+#include "ChiObjectFactory.h"
+
+#include <filesystem>
+
+#define scint static_cast<int>
+
+namespace chi_mesh
+{
+
+RegisterChiObject(chi_mesh, SplitFileMeshGenerator);
+
+// ##################################################################
+chi::InputParameters SplitFileMeshGenerator::GetInputParameters()
+{
+  chi::InputParameters params = MeshGenerator::GetInputParameters();
+
+  params.SetGeneralDescription(
+    "Generates the mesh only on location 0, thereafter partitions the mesh"
+    " but instead of broadcasting the mesh to other locations it creates binary"
+    " mesh files for each location.");
+  params.SetDocGroup("doc_MeshGenerators");
+
+  params.AddOptionalParameter("num_parts",
+                              0,
+                              "The number of parts to generate. If zero will "
+                              "default to the number of MPI processes. Is "
+                              "ignored if the number of MPI processes > 1.");
```

Comment:
Is parts partitions? I think it should be spelt out if it is.

Reply:
Depends on how you look at it. What's the difference between a partition and a part? IDK. Was just trying to keep input language short, but hey, "death to the user!"-Clue.

Renamed to num_partitions

Reply:
Well a part could be a control rod or a fuel rod, i.e. like a physical part you are meshing. Thanks.